tests/src/3d/testqgs4978rendering.cpp

```diff
@@ -0,0 +1,212 @@
+//runs before all tests
+void TestQgs4978Rendering::initTestCase()
+{
+  // init QGIS's paths - true means that all path will be inited from prefix
+  QgsApplication::init();
+  QgsApplication::initQgis();
+  Qgs3D::initialize();
+
+  mReport = QStringLiteral( "<h1>3D Rendering Tests</h1>\n" );
+
+  mProject.reset( new QgsProject );
+  QgsCoordinateReferenceSystem newCrs;
+  newCrs.createFromString( "EPSG:4978" );
+  mProject->setCrs( newCrs );
+}
+
+//runs after all tests
+void TestQgs4978Rendering::cleanupTestCase()
+{
+  mProject.reset();
+
+  const QString myReportFile = QDir::tempPath() + "/qgistest.html";
+  QFile myFile( myReportFile );
+  if ( myFile.open( QIODevice::WriteOnly | QIODevice::Append ) )
+  {
+    QTextStream myQTextStream( &myFile );
+    myQTextStream << mReport;
+    myFile.close();
+  }
+
+  QgsApplication::exitQgis();
+}
+
+
+bool TestQgs4978Rendering::render3dCheck( const QString &testName, QImage &image, int mismatchCount )
+{
+  mReport += "<h2>" + testName + "</h2>\n";
+  const QString myTmpDir = QDir::tempPath() + '/';
+  const QString myFileName = myTmpDir + testName + ".png";
+  image.save( myFileName, "PNG" );
+  QgsMultiRenderChecker myChecker;
+  myChecker.setControlPathPrefix( QStringLiteral( "3d" ) );
+  myChecker.setControlName( "expected_" + testName );
+  myChecker.setRenderedImage( myFileName );
+  myChecker.setColorTolerance( 2 );  // color tolerance < 2 was failing polygon3d_extrusion test
+  const bool myResultFlag = myChecker.runTest( testName, mismatchCount );
+  mReport += myChecker.report();
+  return myResultFlag;
+}
+
+bool TestQgs4978Rendering::render2dCheck( const QString &testName, QgsVectorLayer *layer, QgsRectangle extent, QgsCoordinateReferenceSystem *crs )
+{
+  mReport += "<h2>" + testName + "</h2>\n";
+
+  const QString myTmpDir = QDir::tempPath() + '/';
+  const QString myFileName = myTmpDir + testName + ".png";
+
+  auto mMapSettings = new QgsMapSettings();
+  mMapSettings->setLayers( QList<QgsMapLayer *>() << layer );
+  mMapSettings->setExtent( extent );
+  mMapSettings->setOutputDpi( 96 );
+  if ( crs == nullptr )
+  {
+    QgsCoordinateReferenceSystem newCrs;
+    newCrs.createFromString( "EPSG:3857" );
+    mMapSettings->setDestinationCrs( newCrs );
+  }
+  else
+  {
+    mMapSettings->setDestinationCrs( *crs );
+  }
+
+  QgsRenderChecker myChecker;
+  myChecker.setControlPathPrefix( QStringLiteral( "3d" ) );
+  myChecker.setControlName( "expected_" + testName );
+  myChecker.setMapSettings( *mMapSettings );
+  myChecker.setRenderedImage( myFileName );
+  myChecker.setColorTolerance( 15 );
+  const bool myResultFlag = myChecker.runTest( testName, 0 );
+  mReport += myChecker.report();
+  return myResultFlag;
+}
+
+void TestQgs4978Rendering::test2dLineRendering()
+{
+  QgsVectorLayer *layerLines = new QgsVectorLayer( QString( TEST_DATA_DIR ) + "/3d/earth_size_sphere_4978.gpkg", "lines", "ogr" );
+
+  QgsLineSymbol *fillSymbol = new QgsLineSymbol();
+  fillSymbol->setWidth( 0.3 );
```

Comment:
Can you make these lines much thicker so that the test is more robust against cross platform rendering differences?